auth/env_keychain.go

```diff
@@ -77,32 +98,38 @@ func ReadEnvVar(envVar string) (map[string]string, error) {
 	return authMap, nil
 }
 
-// BuildEnvVar creates the contents to use for authentication environment variable.
-//
-// Complementary to `ReadEnvVar`.
-func BuildEnvVar(keychain authn.Keychain, images ...string) (string, error) {
+func buildAuthMap(keychain authn.Keychain, images ...string) map[string]string {
 	registryAuths := map[string]string{}
 
 	for _, image := range images {
 		reference, authenticator, err := ReferenceForRepoName(keychain, image)
 		if err != nil {
-			return "", nil
+			return map[string]string{}
```

Comment:
This problem existed before this PR but do we want to `continue` here instead of returning empty, in case this succeeds for other images in the set? Same on lines 115 and 120?

Reply:
Great! With this change, we no longer need to strip out empty image names, so it makes the other code simpler too.